src/votekit/elections/election_types.py

```diff
@@ -1310,3 +1312,125 @@ def run_election(self):
             self.run_step()
 
         return self.state
+
+
+class PluralityVeto(Election):
+    """
+    Scores each candidate by their plurality (number of first) place votes,
+    then in a randomized order it lets each voter decrement the score of their
+    least favorite candidate. The candidate with the largest score at the end is
+    then chosen as the winner.
+
+    Args:
+      profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to elect.
+
+    Attributes:
+      _profile (PreferenceProfile): PreferenceProfile to run election on.
+      seats (int): Number of seats to be elected.
+
+    """
+
+    def __init__(self, profile: PreferenceProfile, seats: int):
+        super().__init__(profile, ballot_ties=False)
+        self.seats = seats
+
+    def next_round(self) -> bool:
+        """
+        Determines if another round is needed.
+
+        Returns:
+            (bool) True if number of seats has not been met, False otherwise.
+        """
+        cands_elected = [len(s) for s in self.state.winners()]
+        return sum(cands_elected) < self.seats
+
+    def run_step(self):
+        if self.next_round():
+            candidates = self.state.profile.get_candidates()
+            ballots = self.state.profile.ballots
+
+            # First, count a plurality score for each of the candidates
+            candidate_approvals = {c: Fraction(0) for c in candidates}
+            for ballot in ballots:
+                # From each ballot, randomly choose one candidate from the
+                # set of their first place ranking candidates to accept
+                first_choice = random.choice(list(ballot.ranking[0]))
+                candidate_approvals[first_choice] += ballot.weight
+
+            # Next take a randomized ordering of the ballots:
+            random_order = np.random.permutation(len(ballots))
+
+            # Use another dictionary to keep track of which candidates have
+            # non-zero approval scores
+            A = {c: score for c, score in candidate_approvals.items() if score > 0}
+
+            # Finally, decrement scores with vetos from each of the voters
+            last_eliminated = None
+            for r in random_order:
+                # From each ballot, look at all the remaining candidates
+                # in A, and veto one which this ballot ranks lowest.
+                # This might be expensive, let me know if you have any ideas for how to improve
+                ballot = ballots[r]
+
+                max_rank = -1
+                max_rank_cands = []
+                for c in A.keys():
```

Comment:
If I'm understanding the election correctly, you want to choose a random candidate from the lowest ranking position. I would just access the last ranking position by `ballot.ranking[-1]` and skip this whole for loop.

Reply:
The use of max is also confusing here since I think you want the "worst" ranked candidate. Max could mean "maximally ranked" which could mean a good candidate or something. I would just say "least_preferred" or something.

Reply:
So the trouble I had is that it's not always the very last ranking that I need to find. I keep a set of electable candidates called A from which candidates get eliminated as the process goes on. The loop finds each voters least preferred candidate among those in A. But yes this loop is not efficient and now I'm not sure its the right way to do multi-winner voting, so I'll have think about this and get back to you.